packages/lit-dev-content/src/global/hydrate-common-components.ts

```diff
@@ -63,6 +49,73 @@ const activateMobileDetails = async () => {
   );
 };
 
+// hydrate the search modal
+
+// There are 2 modal buttons on the page. In the header and in the mobile nav.
+// We need to open the one in the current view or else it'll open a hidden
+// modal that makes the dom inert.
+const searchModals = document.querySelectorAll(
+  'litdev-search-modal'
+) as NodeListOf<LitDevSearchModal>;
+let modalHydrated = false;
+
+const hydrateSearchModal = async () => {
+  if (modalHydrated) return;
+  await import('../components/litdev-search-modal.js');
+  modalHydrated = true;
+};
+
+for (const searchModal of searchModals) {
+  searchModal?.addEventListener('focus-within', hydrateSearchModal, {
+    once: true,
+  });
+  searchModal?.addEventListener('pointerenter', hydrateSearchModal, {
+    once: true,
+  });
+
+  searchModal?.addEventListener(
+    'click',
+    () => {
+      searchModal.open = true;
+      hydrateSearchModal();
+    },
+    {
+      once: true,
+    }
+  );
+}
+
+/**
+ * This will listen for the CMD or CTRL+K keypress and open the active search
+ * modal. If the modal is not hydrated it will hydrate it. This event listener
+ * does not get removed because it needs to determine which of the two modals to
+ * open based on the current viewport (mobile or desktop).
+ */
+const hydrateAndOpenOnCMDK = async (e: KeyboardEvent) => {
+  if (e.key === 'k' && (e.ctrlKey || e.metaKey)) {
+    e.preventDefault();
+    const isMobile = window.matchMedia('(max-width: 864px)').matches;
+    const navBar = isMobile ? 'mobileSiteNav' : 'desktopNav';
+
+    // We need to make sure we open the currently visible modal or else it'll
+    // open a hidden modal that makes the dom inert.
+    const searchModal = document.querySelector(
+      `#${navBar} litdev-search-modal`
+    ) as LitDevSearchModal;
```

Comment:
(optional) It may be worth explicitly also handling `undefined` and warning or throwing so that when developing we can quickly pinpoint the issue. Although line 112 would throw a runtime error.

Reply:
Personally I'd rather it throw the error so we know to remove this code if it no longer exists